### ISSUE TITLE
Slow query, 42% of time spent in OverlapsTimeRange

### ISSUE THREAD

**Issue by epa095** [influxdata/influxdb#9476]
### Bug report
__System info:__ 
Influxdb 1.4.3 from docker.

__Steps to reproduce:__
1) Install same influx on two identical machines, (8 cores 32GB RAM) 
2) import the same data on both instances,
3) wait until both have done full compaction,
4) run the same query ("SELECT count(Value) from..") on both machines and time them.

__Expected behavior:__ [What you expected to happen]
Two fast queries.

__Actual behavior:__ [What actually happened]
One (the "fast") answers in 12 seconds, the slow uses 16 minutes. 

__Additional info:__ [Include gist of relevant config, logs, etc.]
Config: 
[data]
  engine = "tsm1"
  cache-snapshot-write-cold-duration = "1m0s"
  compact-full-write-cold-duration = "1m"

- Shard duration is 9000 weeks (only 1 shard).
- iotop/iostat shows no IO on any of the instances. 
- Profiling (see attachment) shows that on the slow machine, 42% of the time is spent in "OverlapsTimeRange", under "nextTSM", while "OverlapsTimeRange" does not show up in the profile of the fast instance. 
- More info can be found in my [forum post](https://community.influxdata.com/t/debug-extremely-slow-querie/4089). 
- Profiles:
[fast-profiles.tar.gz](https://github.com/influxdata/influxdb/files/1753076/fast-profiles.tar.gz)
[slow-profiles.tar.gz](https://github.com/influxdata/influxdb/files/1753077/slow-profiles.tar.gz)

**Comment by jwilder** [comment]
Would you be able to run `influx_inspect dumptsm -all /path/to/tsm/file` on a few of the tsm files in your slow shard and attach the output here?  You can also email it to me if you prefer.

That profile hints that there are a lot of blocks to look at which is a little odd if the shard has been fully compacted.

**Comment by hpbieker** [comment]
@epa095 did you write all data (per measurement/field) in an ascending time order? If not you might have a look at https://github.com/influxdata/influxdb/issues/9255 . This issue should be fixed in master and 1.5 branches.

**Comment by epa095** [comment]
@jwilder : I sent you on email (which I found her on github, hopefully it reached you)
@hpbieker : Good guess, but I did import in ascending order. I saw issue #9255 earlier and got optimistic, so I wrote a small [python program](https://gist.github.com/epa095/21a3f3fb6dc615b887569183f32c84d3) to check if it could be the problem, but no, it looks like the blocks are all consecutive. Both inside each tsm, and across them.
